src/core/server/saved_objects/permission_control/client.ts

```diff
@@ -85,27 +92,51 @@ export class SavedObjectsPermissionControl {
     permissionModes: SavedObjectsPermissionModes
   ) {
     const savedObjectsGet = await this.bulkGetSavedObjects(request, savedObjects);
-    if (savedObjectsGet) {
-      const principals = this.getPrincipalsFromRequest(request);
-      const hasAllPermission = savedObjectsGet.every((item) => {
-        // for object that doesn't contain ACL like config, return true
-        if (!item.permissions) {
-          return true;
-        }
-        const aclInstance = new ACL(item.permissions);
-        return aclInstance.hasPermission(permissionModes, principals);
-      });
+    if (!savedObjectsGet) {
+      return {
+        success: false,
+        error: i18n.translate('savedObjects.permission.notFound', {
+          defaultMessage: 'Can not find target saved objects.',
+        }),
+      };
+    }
+
+    if (savedObjectsGet.length === 1 && !!savedObjectsGet[0].error) {
       return {
-        success: true,
-        result: hasAllPermission,
+        success: false,
+        error: savedObjectsGet[0].error,
       };
     }
 
+    const principals = this.getPrincipalsFromRequest(request);
+    const hasAllPermission = savedObjectsGet.every((item) => {
+      // for object that doesn't contain ACL like config, return true
+      if (!item.permissions) {
+        return true;
+      }
+      const aclInstance = new ACL(item.permissions);
+      return aclInstance.hasPermission(permissionModes, principals);
+    });
+    if (!hasAllPermission) {
+      const savedObjectsBasicInfo = savedObjectsGet.map((item) => {
+        return {
+          id: item.id,
+          type: item.type,
+          workspaces: item.workspaces,
+          permissions: item.permissions,
+        };
+      });
+      this.logger.debug(
+        `Authorization failed, principals: ${JSON.stringify(
+          principals
+        )} has no [${permissionModes}] permissions on the requested saved object: ${JSON.stringify(
+          savedObjectsBasicInfo
+        )}`
+      );
```

Comment:
seems that we log all the saved objects in request here, i am not sure whether we should log the saved objects which user don't have permission or all of them?

Reply:
Only log the saved objects which user has no permission is better, I've change that, please help to take a look.